python/lsst/ts/salobj/topics/remote_command.py

```diff
@@ -469,30 +467,37 @@ async def start(
         salobj.AckTimeoutError
             If the command times out.
         RuntimeError
-            If the ``salinfo`` has not started reading.
+            If the ``salinfo`` is not running.
         TypeError
             If ``data`` is not None and not an instance of `DataType`.
         """
-        self.salinfo.assert_started()
-        if data is not None:
-            self.data = data
-        # else use the existing data, since it may have been set
-        # via a call to "set"
+        self.salinfo.assert_running()
```

Comment:
I think raising RuntimeError is better than AssertionError when a message cannot be sent because the SalInfo is not running.

I prefer to reserve AssertionError for unit tests.